core/src/main/java/org/hibernate/ogm/jpa/impl/OgmEntityManagerFactory.java

```diff
@@ -39,22 +41,26 @@ public OgmEntityManagerFactory(EntityManagerFactory hibernateEmf) {
 
 	@Override
 	public EntityManager createEntityManager() {
-		return new OgmEntityManager( this, hibernateEmf.createEntityManager() );
+		return new OgmEntityManager( this, hibernateEmf.createEntityManager().unwrap( EntityManagerImpl.class ) );
```

Comment:
There is no handling for the specific implementation _EntityManagerImpl.class_ in the unwrap() implementation of our EntityManager.
This happens to work as unwrap() is able to cast it to _EntityManager_ (the interface), but this only works if the actual implementation happens to be an _EntityManagerImpl_. So.. shouldn't you rather cast explicitly?